app/services/task_category_service.rb

```diff
@@ -0,0 +1,40 @@
+class TaskCategoryService
+  attr_accessor :errors
+
+  def find_all
+    TaskCategory.all.to_a
+  end
+
+  def create(params)
+    task_category = TaskCategory.find_by(name: params[:name])
+    if task_category
+      self.errors = 'Task Category Already Exists with this name'
+    else 
+      unless TaskCategory.create!(name: params[:name], description: params[:description])
```

Comment:
this will raise exception on failure, I have explained many times the difference bw normal create and create!, please read carefully once again